presto-router/src/test/java/com/facebook/presto/router/TestRouterAuthentication.java

```diff
@@ -0,0 +1,228 @@
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import javax.ws.rs.core.UriBuilder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+
+import static com.facebook.presto.router.TestingRouterUtil.createPrestoServer;
+import static com.facebook.presto.router.TestingRouterUtil.getConfigFile;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+
+@Test(singleThreaded = true)
+public class TestRouterAuthentication
+{
+    private static final Logger log = Logger.get(TestRouterAuthentication.class);
+    private static final String jksPassword = "testPass";
+
+    private File configFile;
+    private LifeCycleManager lifeCycleManager;
+    private File authenticatorPropertiesFile;
+    private HttpServerInfo httpServerInfo;
+    private Path storePath;
+    private Path keystorePath;
+    private Path truststorePath;
+    private HttpClient httpClient;
+
+    private void runKeytoolCommand(String command, Object... args)
+            throws IOException, InterruptedException
+    {
+        new ProcessBuilder()
+                .directory(storePath.toFile())
+                .command(Splitter.on(" ").splitToStream(format(command, args)).toArray(String[]::new))
+                .start()
+                .waitFor();
+    }
+
+    private void setupCertStores()
+            throws Exception
+    {
+        String alias = "localhost";
+        runKeytoolCommand("keytool -genkeypair -alias %s -keyalg RSA -keysize 2048 -validity 365 -keystore keystore.jks -storepass %s -dname CN=localhost -ext SAN=ip:127.0.0.1", alias, jksPassword);
+        runKeytoolCommand("keytool -exportcert -alias %s -keystore keystore.jks -file certificate.cer -storepass %s", alias, jksPassword);
+        runKeytoolCommand("keytool -importcert -alias %s -file certificate.cer -keystore truststore.jks -storepass %s -noprompt", alias, jksPassword);
+        keystorePath = storePath.resolve("keystore.jks");
+        truststorePath = storePath.resolve("truststore.jks");
+    }
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        Logging.initialize();
+        Path tempFile = Files.createTempFile("temp-config", ".json");
+        configFile = getConfigFile(ImmutableList.of(createPrestoServer()), tempFile.toFile());
+
+        authenticatorPropertiesFile = new File("etc/password-authenticator.properties");
+        Path passwordFilePath = Files.createTempFile("passwords", ".db");
+        // credentials come from htpasswd -n -B -C 8, input "testpass"
+        Files.write(passwordFilePath, "testuser:$2y$08$KBfSimK6KTZFyCKlJACpTu7VMBHlnFixXm8tDh9I0rDf3IIuobtHy".getBytes(UTF_8));
```

Comment:
Can we generate pass here, instead of hard coding ?

Reply:
No this can't be generated easily because it requires an algorithm to encrypt the password which isn't easily accessible. I think this is fine

Reply:
Adding the comment, how it was generated should be good enough in that case.